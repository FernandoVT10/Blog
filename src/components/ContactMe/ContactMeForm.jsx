import ValidationInput from "../form/ValidationInput";
import Api from "../../ApiController";

import { useState } from "react";

export default ({ setModalMessage, setLoading }) => {
    const [email, setEmail] = useState({ value: "", valid: false });
    const [message, setMessage] = useState("");

    const sendMessage = e => {
        e.preventDefault();
        e.target.reset();

        if(email.valid) {
            setLoading(true);

            Api.post("messages/addMessage", { email: email.value, message })
            .then(data => {
                if(data.status) {
                    setModalMessage(data.message);
                } else {
                    setModalMessage("An error has occurred");
                }

                setLoading(false);
            });
        }
    };

    return (
        <form onSubmit={sendMessage}>
            <h2 className="title mb-3">Contact Me</h2>

            <ValidationInput
            type="email"
            placeholder="Enter your email"
            onChange={setEmail} />

            <textarea
            className="formulary__textarea mt-3"
            placeholder="Enter your message"
            onChange={({ target: { value } }) => setMessage(value)}
            maxLength="500"
            rows="5"
            required></textarea>

            <button className="submit-button mt-3">
                Send Message
            </button>
        </form>
    );
}