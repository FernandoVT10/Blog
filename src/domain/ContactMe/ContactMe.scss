.contact-me {
    &__about-me {
        margin-top: 20px;
        padding: 25px 15px;
        background: var(--primary);
    }

    &__title {
        font-size: 1.5em;
        color: var(--white);
        font-weight: bold;
    }

    &__profile-container {
        position: relative;
        margin-bottom: 85px;
    }

    &__cover {
        background: url(http://localhost:3000/img/cover.jpg);
        background-size: cover;
        background-position: center;
        border: solid 3px var(--secondary);
        width: 100%;
        height: 250px;
    }

    &__picture {
        position: absolute;
        bottom: -75px;
        left: 20px;
        width: 150px;
        border: solid 3px var(--secondary);
        border-radius: 50%;
    }

    &__description {
        color: var(--white);
        font-weight: 500;
        font-size: 18px;
        line-height: 3;
        margin-top: 25px;
        margin-bottom: 0;
    }

    &__form {
        margin-top: 20px;
        padding: 20px;
        background: var(--primary);
    }
}