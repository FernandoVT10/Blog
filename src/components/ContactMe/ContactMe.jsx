import Modal from "../Modal";
import FullScreenLoader from "../FullScreenLoader";

import { useState } from "react";

import "./ContactMe.scss";
import ContactMeForm from "./ContactMeForm";

export default () => {
    const [modalMessage, setModalMessage] = useState("");
    const [loading, setLoading] = useState(false);

    return (
        <div>
            <FullScreenLoader loading={loading}/>

            <Modal
            title="Message status"
            active={modalMessage.length > 0}
            onClose={() => setModalMessage("")}>
                <p>{ modalMessage }</p>
            </Modal>

            <div className="body">
                <div className="container-fluid contact-me">
                    <div className="row">
                        <div className="col-12 contact-me__who-is mt-5">
                            <h2 className="title">Who is Fernando Vaca Tamayo?</h2>
                            <p className="contact-me__description">
                                Lorem ipsum dolor sit amet consectetur adipisicing elit. 
                                Pariatur, tenetur fugiat accusamus atque reprehenderit ratione
                                magni adipisci quos, quod laborum, hic maxime mollitia voluptatum 
                                a nobis at! Dolorem et nemo id aspernatur doloremque minima,
                                incidunt repellendus, fuga aperiam ut qui consequatur laboriosam, 
                                cumque rem vero magnam.
                            </p>
                        </div>
                    </div>

                    <div className="row contact-me__form mt-4">
                        <div className="col-12 col-lg-8">
                            <ContactMeForm
                            setLoading={setLoading} 
                            setModalMessage={setModalMessage} />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}