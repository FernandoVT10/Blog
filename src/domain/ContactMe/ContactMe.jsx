import ContactMeForm from "./ContactMeForm/ContactMeForm";

import "./ContactMe.scss";

export default () => {
    return (
        <div className="contact-me">
            <div className="body">
                <div className="container-fluid">
                    <div className="row">

                        <div className="col-12 contact-me__about-me">
                            <div className="contact-me__profile-container">
                                <div className="contact-me__cover"></div>
                                <img
                                className="contact-me__picture"
                                src="/img/profile-picture.jpg"
                                alt="My profile picture"/>
                            </div>

                            <h2 className="contact-me__title">About Fernando Vaca Tamayo</h2>
                            <p className="contact-me__description">
                                <b>Hello!</b>. I'm a mexican web developer.<br/>

                                I've been studying technology for more than 4 years.<br/>

                                Do i know <b>english</b> at a normal level?...
                                Well i can write and read.
                                I'm practicing <b>listening</b> and <b>speaking</b>.<br/>

                                I'm a little bad at design, but i'm <b>working</b> to fix it.<br/>

                                The language i like the most are <b>JavaScript</b>.<br/>

                                Now i'm studying <b>TypeScript</b>.<br/>

                                I really like to see <b>clean code</b>,
                                i find it so beautiful and fun.<br/>

                                And to finish a curious fact about me. I discovered the programming
                                thanks to <b>Minecraft</b> and it's <b>mods</b>.
                            </p>
                        </div>

                        <div className="col-lg-6 contact-me__form">
                            <ContactMeForm />
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}