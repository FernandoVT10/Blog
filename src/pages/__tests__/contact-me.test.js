import { act, Simulate } from "react-dom/test-utils";
import { render } from "react-dom";
import ContactMe from "../contact-me";

let container;

describe("Contact Me page", () => {
    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    
    afterEach(() => {
        document.body.removeChild(container);
        container = null;
    });

    it('Check if renders correctly', () => {
        render(<ContactMe/>, container);

        expect(container).toMatchSnapshot();
    });

    it("It should show the modal with a message", async () => {
        render(<ContactMe/>, container);

        const input = container.querySelector("input");
        const textarea = container.querySelector("textarea");

        act(() => {
            input.value = "test@gmail.com";
            textarea.value = "Test message";

            Simulate.change(input);
            Simulate.change(textarea);
        });

        fetchMock.doMock();
        fetchMock.mockOnce(JSON.stringify({
            status: true,
            message: "Test message"
        }));

        const form = container.querySelector("form");

        await act(async () => {
            Simulate.submit(form);
        });

        const p = container.querySelector(".modal-body p");

        expect(p.textContent).toBe("Test message");
    });
});