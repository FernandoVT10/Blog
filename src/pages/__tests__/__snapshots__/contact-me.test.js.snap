// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Me page Check if renders correctly 1`] = `
<div>
  <div>
    <header>
      <nav
        class="navbar navbar-expand-md navbar-dark"
      >
        <a
          class="navbar-brand navbar__brand"
          href="/"
        >
          <img
            alt="Icon"
            class="d-inline-block align-top rounded-circle"
            height="50"
            src="/favicon.ico"
            width="50"
          />
          <h1
            class="navbar__title"
          >
            Fernando Vaca Tamayo
          </h1>
        </a>
        <button
          aria-controls="main-navbar"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-target="#main-navbar"
          data-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse navbar__items"
          id="main-navbar"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link navbar__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link navbar__link"
                href="/articles"
              >
                Articles
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link navbar__link"
                href="/projects"
              >
                Projects
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link navbar__link"
                href="/contact-me"
              >
                Contact me
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
    <div
      class="full-screen-loader "
    >
      <div
        class="full-screen-loader__spinner spinner-border text-light"
        role="status"
      >
        <span
          class="sr-only"
        >
          Loading...
        </span>
      </div>
      <h3
        class="full-screen-loader__text"
      >
        Loading...
      </h3>
    </div>
    <div
      class="modal fade"
      id="-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog "
        role="document"
      >
        <div
          class="modal-content modal__content"
        >
          <div
            class="modal-header modal__header"
          >
            <h5
              class="modal__title"
            >
              Message status
            </h5>
            <button
              aria-label="Close"
              class="close modal__button"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            class="modal-body modal__body"
          >
            <p />
          </div>
        </div>
      </div>
    </div>
    <div
      class="body"
    >
      <div
        class="container-fluid contact-me"
      >
        <div
          class="row"
        >
          <div
            class="col-12 contact-me__who-is mt-5"
          >
            <h2
              class="title"
            >
              Who is Fernando Vaca Tamayo?
            </h2>
            <p
              class="contact-me__description"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Pariatur, tenetur fugiat accusamus atque reprehenderit ratione magni adipisci quos, quod laborum, hic maxime mollitia voluptatum a nobis at! Dolorem et nemo id aspernatur doloremque minima, incidunt repellendus, fuga aperiam ut qui consequatur laboriosam, cumque rem vero magnam.
              <br />
              <br />
              Harum nihil eius atque ratione at est vero. Magni laudantium culpa earum magnam aliquid, debitis eaque fugit eos praesentium deleniti rem sapiente sint at voluptatibus eligendi repellendus repellat tempora unde sed sequi explicabo temporibus perferendis. Sint quo dolorem eos animi sapiente. Totam quisquam velit mollitia cum eaque porro dolorem pariatur, iusto, nostrum incidunt nesciunt.
              <br />
              <br />
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Pariatur, tenetur fugiat accusamus atque reprehenderit ratione magni adipisci quos, quod laborum, hic maxime mollitia voluptatum a nobis at! Dolorem et nemo id aspernatur doloremque minima, incidunt repellendus, fuga aperiam ut qui consequatur laboriosam, cumque rem vero magnam.
              <br />
              <br />
              Harum nihil eius atque ratione at est vero. Magni laudantium culpa earum magnam aliquid, debitis eaque fugit eos praesentium deleniti rem sapiente sint at voluptatibus eligendi repellendus repellat tempora unde sed sequi explicabo temporibus perferendis. Sint quo dolorem eos animi sapiente. Totam quisquam velit mollitia cum eaque porro dolorem pariatur, iusto, nostrum incidunt nesciunt.
            </p>
          </div>
        </div>
        <div
          class="row contact-me__form mt-4"
        >
          <div
            class="col-6 col-lg-8"
          >
            <form>
              <h2
                class="title mb-3"
              >
                Contact Me
              </h2>
              <div
                class="formulary__input-container"
              >
                <input
                  autocomplete="email"
                  class="formulary__input "
                  id="-input"
                  maxlength="100"
                  placeholder="Enter your email"
                  required=""
                  type="email"
                />
                <label
                  class="formulary__error-label mt-2 "
                  for="-input"
                >
                  <i
                    aria-hidden="true"
                    class="fas fa-info-circle mr-1"
                  />
                  
                </label>
              </div>
              <textarea
                class="formulary__textarea mt-3"
                maxlength="500"
                placeholder="Enter your message"
                required=""
                rows="5"
              />
              <button
                class="submit-button mt-3"
              >
                Send Message
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="main-footer container-fluid mt-4"
    >
      <div
        class="full-screen-loader "
      >
        <div
          class="full-screen-loader__spinner spinner-border text-light"
          role="status"
        >
          <span
            class="sr-only"
          >
            Loading...
          </span>
        </div>
        <h3
          class="full-screen-loader__text"
        >
          Loading...
        </h3>
      </div>
      <div
        class="modal fade"
        id="-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog "
          role="document"
        >
          <div
            class="modal-content modal__content"
          >
            <div
              class="modal-header modal__header"
            >
              <h5
                class="modal__title"
              >
                Suscription confirm status
              </h5>
              <button
                aria-label="Close"
                class="close modal__button"
                data-dismiss="modal"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
            <div
              class="modal-body modal__body"
            >
              <p />
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12 col-sm-6 col-xl-3"
        >
          <a
            class="main-footer__brand"
            href="/"
          >
            <img
              alt="Icon"
              class="d-inline-block align-top rounded-circle"
              height="50"
              src="/favicon.ico"
              width="50"
            />
            <h3
              class="main-footer__title"
            >
              Fernando Vaca Tamayo
            </h3>
          </a>
          <div
            class="mt-4"
          >
            <h3
              class="main-footer__subtitle"
            >
              Contact Me
            </h3>
            <p>
              <i
                aria-hidden="true"
                class="fas fa-envelope main-footer__icon"
              />
              <span
                class="main-footer__text"
              >
                vacatamayo321@gmail.com
              </span>
            </p>
          </div>
        </div>
        <div
          class="col-12 col-sm-6 col-xl-3"
        >
          <h3
            class="main-footer__subtitle"
          >
            Social Networks
          </h3>
          <a
            href="https://twitter.com/FernandoVT10"
            target="_blank"
          >
            <p>
              <i
                aria-hidden="true"
                class="fab fa-twitter main-footer__icon"
              />
              <span
                class="main-footer__text"
              >
                @FernandoVT10
              </span>
            </p>
          </a>
          <a
            href="https://github.com/FernandoVT10"
            target="_blank"
          >
            <p>
              <i
                aria-hidden="true"
                class="fab fa-github main-footer__icon"
              />
              <span
                class="main-footer__text"
              >
                FernandoVT10
              </span>
            </p>
          </a>
        </div>
        <div
          class="col-12 col-sm-6 col-xl-3"
        >
          <h3
            class="main-footer__subtitle"
          >
            Description
          </h3>
          <p
            class="main-footer__description"
          >
            This blog is my personal blog xD
          </p>
        </div>
        <div
          class="col-12 col-sm-6 col-xl-3"
        >
          <h3
            class="main-footer__subtitle"
          >
            Suscribe to my blog
          </h3>
          <form>
            <div
              class="formulary__input-container"
            >
              <input
                autocomplete="email"
                class="formulary__input "
                id="-input"
                maxlength="100"
                placeholder="Enter your email"
                required=""
                type="email"
              />
              <label
                class="formulary__error-label mt-2 "
                for="-input"
              >
                <i
                  aria-hidden="true"
                  class="fas fa-info-circle mr-1"
                />
                
              </label>
            </div>
            <button
              class="submit-button mt-3"
            >
              Suscribe
            </button>
          </form>
        </div>
      </div>
    </footer>
    <script
      src="https://cdn.jsdelivr.net/npm/chart.js@2.9.3/dist/Chart.min.js"
    />
    <script
      src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
    />
    <script
      src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
    />
    <script
      src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
    />
    <script
      src="https://kit.fontawesome.com/63ef8f1397.js"
    />
  </div>
</div>
`;
